/*
    CF 微信对接协议
    RPC名称:cf.wechat.rpc
	Exchange:cf.wechat.notify
*/
package cf.wechat;

//-------------------------全部消息
message Header{
	required string sender=1;
	required string sender_type=2;
	optional uint32 invoke_id=3;
}
message Message{
	// 头部
	required Header header=1;
	// 主体
	oneof content{
		// 请求:获取用户基本信息
		UserInfoRequest req_userinfo_query=2;
		// 回应:获取用户基本信息
		UserInfoResponse res_userinfo_query=3;	
		// 请求:获取全部会员openid列表
                SyncUserInfoRequest req_userinfo_sync=4;
                // 回应:获取全部会员openid列表
                SyncUserInfoResponse res_userinfo_sync=5;
	}
}

// 客户检索条件
message CustomerSearchCondition
{
	// 客户ID
	optional string id=10;
	// 微信相关
	optional string cf_account_id=11;
	optional string wechat_open_id=12;
	optional string wechat_union_id=13;
	optional string code=18;
	// Ali账号
	optional string ali_id=14;
	// 手机号码
	optional string mobile=15;
	// 设备MAC地址
	optional string device_mac=16;
	// Email地址
	optional string email=17;
}


// 请求:获取用户基本信息
message UserInfoRequest
{
	// 发送账号ID
	required string account_id=1;
	// 用户，最终需要openid,单个或批量获取
	repeated CustomerSearchCondition customer_search_condition=2;
	// 语言，默认CN
	optional ELanguage language=3;
}
// 回应:获取用户基本信息
message UserInfoResponse
{
	// 错误码
	required int32 errcode=1;
	// 错误消息
	optional string errmsg=2;
	// 发送结果
	repeated UserInfo result=3;
	message UserInfo
	{
		// 关注状态
		required bool subscribe=20;
		// 将用户openid返回，多条数据时用于区分
		required string openid=21;
		// 昵称
		required string nickname=22;
		// 性别，1是男性2是女性0是未知
		required int32 sex=23;
		// 国家
		required string country=25;
		// 省份
		required string province=26;
		// 城市
		required string city=24;
		// 用户头像资源uuid
		required string avatar_uuid=28;
		// 关注时间(UTC)
		required uint64 subscribe_time=29;
		// 全局ID
		optional string union_id=30;
		// 备注
		required string remark=31;
		// 所属组ID，暂不支持
		optional int32 group_id=32;
	}
}

// 请求:获取全部会员openid列表
message SyncUserInfoRequest
{
	// 发送账号ID
	required string account_id=1;
	// 用户，最终需要openid,单个或批量获取
	optional CustomerSearchCondition customer_search_condition=2;
	// 语言，默认CN
	optional ELanguage language=3;
}

// 回应:获取全部会员openid列表
message SyncUserInfoResponse
{
	// 错误码
	required int32 errcode=1;
	// 错误消息
	optional string errmsg=2;
	// 发送结果
	repeated string openid_list=3;
}

// 语言
enum ELanguage
{
	EN=1;
	CN=2;
	TW=3;
}